overlord/state/taskrunner.go

```diff
@@ -258,7 +258,9 @@ func (r *TaskRunner) run(t *Task) {
 				// Would work without it but might take two ensures.
 				r.tryUndo(t)
 			} else {
-				t.SetStatus(WaitStatus)
+				// XXX: Should we allow supplying a state with the Wait error?
+				// Currently using DoneStatus as that was the previous behaviour.
+				t.Wait(DoneStatus)
```

Comment:
I don't know what we need in future code, but in theory we should support at least DoneStatus and UndoneStatus ?

Reply:
Support for this added!